lib/vagrant/cli.rb

```diff
@@ -62,9 +62,11 @@ def execute
       # Initialize and execute the command class, returning the exit status.
       result = 0
       begin
-        @triggers.fire_triggers(@sub_command.to_sym, :before, nil, :command) if Vagrant::Util::Experimental.feature_enabled?("typed_triggers")
+        @triggers.fire(@sub_command, :before, nil, :command) if
```

Comment:
Why are these triggers still fired directly (in contrast to Vagrant::Machine#action)?

Reply:
These are :snowflake: triggers and must be implemented differently to ensure they work as expected. For all the other trigger type integrations, we know that a call stack will be executed from an environment. For commands, while they generally will follow that behavior, it is not guaranteed. Because of this, the triggers for commands need to explicitly wrap the command executions to ensure they are actually called.

Reply:
I seem to get warnings now, even on commands unrelated to what is defined, around these triggers :thinking: 

```
==> ubuntu: The command 'halt' was not found for this trigger.
```

```ruby
 # Command triggers are allowed to run before or after an entire Vagrant command.
  config.trigger.before :halt, type: :command do |t|
    t.info = "Halting machine..."
  end
```

This showed up for when I ran `vagrant up`. I'm guessing this is because I didn't have the experimental flag enabled. It just reads a bit weird for me, is all! Previous to these changes nothing would be printed, which is maybe worse :sweat_smile: 

But now it seems that these turn into command and action triggers, even though I explicitly defined them as a command

```ruby
brian@localghost:triggers % VAGRANT_EXPERIMENTAL="1" be vagrant halt                                     ±[●][master]
==> vagrant: You have enabled the experimental flag with all features enabled.
==> vagrant: Please use with caution, as some of the features may not be fully
==> vagrant: functional yet.
==> vagrant: Running command triggers before halt ...
==> vagrant: Running trigger...
==> vagrant: Halting machine...
==> vagrant: Running action triggers before halt ...
==> vagrant: Running trigger...
==> vagrant: Hello world!!
    vagrant: Running local: Inline script
    vagrant: echo 'Hi'
    vagrant: Hi
    vagrant:
==> ubuntu: The command 'halt' was not found for this trigger.
==> ubuntu: Attempting graceful shutdown of VM...
```

And then I get the warning about the halt trigger not being found, even though it had been ran already.

Reply:
This ended up being just a type casting issue